hls-plugin-api/src/Ide/Plugin/Error.hs

```diff
@@ -0,0 +1,163 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE LambdaCase                #-}
+{-# LANGUAGE OverloadedStrings         #-}
+module Ide.Plugin.Error (
+      -- * Plugin Error Handling API
+    PluginError(..),
+    runExceptT,
+    withExceptT,
+    pluginResponseM,
+    hoistExceptT,
+    hoistMaybeT,
+    handleMaybe,
+    handleMaybeM,
+    getNormalizedFilePathE,
+) where
+
+import           Control.Monad.Extra         (maybeM)
+import           Control.Monad.IO.Class      (MonadIO, liftIO)
+import           Control.Monad.Trans.Class   (lift)
+import           Control.Monad.Trans.Except  (ExceptT (..), mapExceptT,
+                                              runExceptT, throwE, withExceptT)
+import           Control.Monad.Trans.Maybe   (MaybeT, mapMaybeT)
+import qualified Data.Text                   as T
+import           Language.LSP.Protocol.Types
+import           Prettyprinter
+
+-- ----------------------------------------------------------------------------
+-- Plugin Error wrapping
+-- ----------------------------------------------------------------------------
+
+pluginResponseM :: Monad m => (t -> m (Either a b)) -> ExceptT t m b -> m (Either a b)
```

Comment:
I'm sceptical of this function: I think it would be just as clear to inline it! Because it's just `runExceptT . handleError handler`, right? Generally I think we shouldn't be afraid of letting people work with `ExceptT`/`MonadError` directly, they're very familiar interfaces.